### PR TITLE
Feature/kenburns refactor

### PR DESCRIPTION
**Pull request by hessnd**
Will add in unit testing and end to end testing in next feature

### CODE REVIEW

**Comment by nickdandakis** [comment]
Nice one @hessnd.

I'd recommend using [imperative form when writing commit messages](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53). That `really big commit fixing a lot of stuff` killed me.

I like the code splitting and better naming of the store action.

Why aren't y'all using Nuxt again?

**Comment by hessnd** [comment]
@nickdandakis haha you caught me being lazy with that commit message since I had to run and go let the dog out.

Yea tried to update as much of this stuff to how I have been doing setup in other projects.

I will most likely use Nuxt for the next vue project, but wanted to just update this for fun.

File: template/build/check-versions.js [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+const packageConfig = require('../package.json');
+const shell = require('shelljs');
+
+function exec (cmd) {
   return require('child_process')
     .execSync(cmd).toString().trim()
 }
 
-var versionRequirements = [
+const versionRequirements = [
```
Comment (brittneykernan) [inline template/build/check-versions.js:12]: Should cost be let if the variable is modified on line 21?

File: template/build/check-versions.js [hunk 2]
```diff
@@ -1,28 +1,35 @@
...
-  }
-]
+  });
+}
 
 module.exports = function () {
   var warnings = []
-  for (var i = 0; i < versionRequirements.length; i++) {
-    var mod = versionRequirements[i]
+  for (let i = 0; i < versionRequirements.length; i++) {
```
Comment (brittneykernan) [inline template/build/check-versions.js:30]: No action here, but this is interesting re: cost of var vs let in for loop: https://stackoverflow.com/a/40070682

File: template/server.js [hunk 3]
```diff
@@ -1,106 +1,110 @@
...
+
+  res.setHeader("Content-Type", "text/html");
+  res.setHeader("Server", serverInfo);
+
+  const handleError = err => {
+    if (err.url) {
+      res.redirect(err.url)
+    } else if (err.code === 404) {
+      res.status(404).send('404 | Page Not Found');
+    } else {
```
Comment (brittneykernan) [inline template/server.js:79]: I suggest setting up templates for custom non verbose 404/500 errors - clients often expect them designed out.

File: template/server.js [hunk 4]
```diff
@@ -1,106 +1,110 @@
...
-        response.write(
-          `<script>
-             window.INITIAL_STATE = ${serialize(context.initialState, { isJSON: true })};
-           </script>`
-        );
-      }
-
-      firstChunk = false;
+  const context = {
+    title: 'Big Spaceship Vue Template',
```
Comment (brittneykernan) [inline template/server.js:88]: Can/should the title be set when you are spinning up a new project (like how Yeoman does it) or in a more central config file?

File: template/server.js [hunk 5]
```diff
@@ -1,106 +1,110 @@
...
 
-  renderStream.on('error', error => { throw error });
+app.get('*', isProd ? render : (req, res) => {
+  readyPromise.then(() => render(req, res));
 });
 
 const port = process.env.PORT || 8080;
-application.listen(port, () => {
+app.listen(port, () => {
   const uri = `http://localhost:${port}`;
```
Comment (brittneykernan) [inline template/server.js:108]: We could set up our locals to run over https (spoofed) so we're removing that variable difference from environments.

File: template/src/components/Gif.vue [hunk 6]
```diff
@@ -5,13 +5,8 @@
...
-
 export default {
-  created() {
-    const tag = 'big+spaceship';
-    this.$store.dispatch('fetchGif', {tag});
-  },
+  name: 'gif',
   computed: {
     gif() {
       return this.$store.state.gif;
```
Comment (brittneykernan) [inline template/src/components/Gif.vue:12]: Do we need this file?

File: template/src/index.template.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
```
Comment (brittneykernan) [inline template/src/index.template.html:2]: We should always have a lang attribute here. Value should be determined by country site is for. I'd add "en-us" value as default.

File: template/src/index.template.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  <head>
+    <title>{{ title }}</title>
+    <meta charset="utf-8">
```
Comment (brittneykernan) [inline template/src/index.template.html:5]: Research if we still need to add this, I think we do...
<meta http-equiv="X-UA-Compatible" content="IE=edge">

File: template/src/index.template.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    <meta name="mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-status-bar-style" content="default">
+    <link rel="apple-touch-icon" sizes="120x120" href="/public/logo-120.png">
+    <meta name="viewport" content="width=device-width, initial-scale=1, minimal-ui">
+    <link rel="shortcut icon" sizes="48x48" href="/public/logo-48.png">
+    <meta name="theme-color" content="#ea4235">
+    <link rel="manifest" href="/manifest.json">
```
Comment (brittneykernan) [inline template/src/index.template.html:13]: We should always have minimal required og: and tw: meta. Perhaps these live in an include. It's best to use Twitter large summary card so OG image can be reused for Twitter, and include og: img height and width values to precache imagery.
Example project: https://github.com/bigspaceship/sbux-blonde/blob/113d645b563ba8819748e3ffc01e33e9be051569/layouts/main-layout/index.js#L79

Comment (brittneykernan) [reply template/src/index.template.html:13]: Add in meta description too. Still think you need keywords even tho Google doesn't use them.

Comment (brittneykernan) [reply template/src/index.template.html:13]: I'd also want to build in the anti clickjacking script that @paul-baranowski added from sbux project into his nextjs. 

Should add GA/GTM placeholders too in Google recommended locations.

**Comment by brittneykernan** [comment]
Nick, looks good. I can't speak to the server side code, but added some FE suggestions outside of this refactor's remit to incorporate later.

**Review by brittneykernan — APPROVED** [review]

**Review by paul-baranowski — APPROVED** [review]

**Pull request merged**